engines/zypper_auth/lib/zypper_auth/engine.rb

```diff
@@ -20,15 +25,23 @@ def verify_instance(request, logger, system)
         instance_data
       )
 
-      is_valid = verification_provider.instance_valid?
+      verification_provider.instance_valid?
```

Comment:
This function has a side effect now, of assigning the _instance id_ and _billing info_; consider changing the function name, or breaking that into a separate function (e.g. `InstanceVerification#set_instance_details` or something.)

I had to scratch my head for a while to figure out why this function that returns a boolean was being run and the result not assigned. :thinking:

Reply:
The implementation of `set_instance_details` would be CSP-specific and would need to be implemented in every plugin, it would also require re-parsing/re-validating the IID. So for simplicity of the plugin interface it just assigns those attributes in one go (they are needed for logging only and it's OK if they are uninitialized).

The method still returns a boolean, we'll go back to checking it again at some point. :slightly_smiling_face: